test/runtests.jl

```diff
@@ -561,6 +561,10 @@ txt = """
 # https://github.com/quinnj/JSON3.jl/issues/8
 @test eltype(JSON3.read("[1.2, 2.0]")) === Float64
 @test eltype(JSON3.read("[1.2, 2.0, 3.3]")) === Float64
+@test eltype(JSON3.read("[1, 2]")) == Int64
+@test eltype(JSON3.read("[1, 2.3]")) == Float64
```

Comment:
This case is problematic; the problem is that parsing already parsed `1` as an `Int64`, so we can't have the final `eltype` be `Float64` or it will reinterpret the first element wrong (it will reinterpret the bits as `Float64` when the bits are encoded as `Int64`).